wallet/src/ui/app/pages/dapp-tx-approval/index.tsx

```diff
@@ -67,56 +184,80 @@ export function DappTxApprovalPage() {
-                          content: toList(txRequest.tx.arguments),
-                      },
-                      {
-                          label: 'Type arguments',
-                          content: toList(txRequest.tx.typeArguments),
-                      },
-                      { label: 'Gas budget', content: txRequest.tx.gasBudget },
+                      { label: 'Gas Fees', content: txRequest.tx.gasBudget },
                   ]
                 : [
                       {
-                          label: 'Transaction type',
+                          label: 'Transaction Type',
                           content: 'SerializedMoveCall',
                       },
                       { label: 'Contents', content: txRequest?.txBytes },
                   ],
         [txRequest]
     );
+
     return (
         <Loading loading={loading}>
             {txRequest ? (
                 <UserApproveContainer
-                    title="Transaction Request"
                     origin={txRequest.origin}
                     originFavIcon={txRequest.originFavIcon}
                     approveTitle="Approve"
                     rejectTitle="Reject"
                     onSubmit={handleOnSubmit}
                 >
-                    {valuesContent.map(({ label, content }) => (
-                        <Fragment key={label}>
-                            <label className={stUserApprove.label}>
-                                {label}
-                            </label>
-                            <div className={st.value}>{content}</div>
-                        </Fragment>
-                    ))}
+                    <dl className={st.card}>
+                        <div className={st.content}>
+                            {valuesContent.map(({ label, content }) => (
+                                <div key={label} className={st.row}>
+                                    <dt>{label}</dt>
+                                    <dd>{content}</dd>
+                                </div>
+                            ))}
+                        </div>
+                    </dl>
+                    {metadata && tab && (
+                        <>
+                            <div className={st.tabs}>
+                                {Object.entries(metadata).map(
+                                    ([key, value]) =>
+                                        value.children.length > 0 && (
+                                            <button
+                                                type="button"
+                                                className={cl(
+                                                    st.tab,
+                                                    tab === key && st.active
+                                                )}
+                                                // eslint-disable-next-line react/jsx-no-bind
+                                                onClick={() => {
```

Comment:
Why not use the dataset of the button to store the key and use it to switch the tab? instead of disabling eslint (Not a problem here but I think we should avoid it in general)

Reply:
I generally don’t love datasets as a solution. They don’t work for non-host components, so I generally prefer solutions that work consistently. IMO this lint rule should entirely be disabled. It’s true that this pattern can lead to occasional over-updates on memoized components, but those performance hot paths can be solved separately.

Reply:
I generally don’t love datasets as a solution. They don’t work for non-host components, so I generally prefer solutions that work consistently. IMO this lint rule should entirely be disabled. It’s true that this pattern can lead to occasional over-updates on memoized components, but those performance hot paths can be solved separately.

Reply:
I understand, but to be honest I would prefer to be safe and avoid to have to look for those places that might cause performance issues in the future. What case of non-host components do you have in mind?

Reply:
Non-host components are just any non-DOM components. In those cases datasets can't be used, you'd need to use an inline handler like this.

I get your concern for performance, but in my experience inline handlers have _almost never_ been the cause of poor performance in an app. There's a lot of common pitfalls, and generally my approach to performance is to find the actual things that are causing performance issues and address those, rather than taking a defensive approach. I'd be happy to set up some time to talk more through this specific case (inline handlers & memoization), and more generic performance within React.

Reply:
> Non-host components are just any non-DOM components. In those cases datasets can't be used, you'd need to use an inline handler like this.

Yeah, my point was if you see a case of non-host components that we will use, but I get your point.

>I'd be happy to set up some time to talk more through this specific case (inline handlers & memoization), and more generic performance within React

Sure happy to talk! For me not allowing inline arrow functions or binding doesn't cause a lot of overhead and can save us from the trouble later.

Reply:
Oh, I really hope that we end up using a lot of non-host components! In general I'm a big fan of granular components, so I'd love to see more and more of the app become just compositions of lower-level components. As that happens, architectural approaches like data-attributes tend to stop working.

I'll follow-up and set some time up!